Add a System Test for the DepolarizedAnalyserTransmission Algorithm

cailafinn opened the pull request:
### Description of work

#### Summary of work
Add a system test to check the algorithm works consistently across OS' with real data files. 


Fixes #37120 

#### Further detail of work
Also adds the associated data files for running the algorithm properly. The code is set to ignore the error in the fit being quite high. This is intentional. 

### To test:
- Run the algorithm locally using the `./build/systemtest -R DepolarizedAnalyserTransmission` script. 
   - This should download the testing data automatically, but re-running cmake and building the `SystemTestData` target manually may be required. 
- The algorithm should pass without issues. 

*This does not require release notes* because **there are no user-facing changes.**

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

cailafinn committed:
Add Boilerplate system test code

RE #37120

cailafinn committed:
Add basic test implementation

RE #37120

cailafinn committed:
Implement validation

RE #37120

cailafinn committed:
Remove incorrect result check

RE #37120

cailafinn committed:
Upload data files to remote store

RE #37120

jhaigh0 commented on Testing/SystemTests/tests/framework/PolarizationCorrections/DepolarizedAnalyserTransmissionTest.py:
```diff
@@ -0,0 +1,62 @@
+        reference_params = "DepolParamsReference.nxs"
+
+        def validate_group(result, reference):
+            value_names = [result, reference]
+
+            Load(Filename=reference, OutputWorkspace=reference)
+            compare_alg = AlgorithmManager.create("CompareWorkspaces")
+            compare_alg.setPropertyValue("Workspace1", value_names[0])
+            compare_alg.setPropertyValue("Workspace2", value_names[1])
+            compare_alg.setPropertyValue("Tolerance", str(self.tolerance))
+
+            compare_alg.execute()
+            if compare_alg.getPropertyValue("Result") != "1":
+                print(" Workspaces do not match.")
+                print(self.__class__.__name__)
+                SaveNexus(InputWorkspace=value_names[0], Filename=self.__class__.__name__ + "-mismatch.nxs")
+                return False
```
```suggestion
            Load(Filename=reference, OutputWorkspace=reference)
            compare_alg = AlgorithmManager.create("CompareWorkspaces")
            compare_alg.setPropertyValue("Workspace1", result)
            compare_alg.setPropertyValue("Workspace2", reference)
            compare_alg.setPropertyValue("Tolerance", str(self.tolerance))

            compare_alg.execute()
            if compare_alg.getPropertyValue("Result") != "1":
                print(" Workspaces do not match.")
                print(self.__class__.__name__)
                SaveNexus(InputWorkspace=result, Filename=self.__class__.__name__ + "-mismatch.nxs")
                return False
```
Any reason not to just do this? Sorry to be picky

jhaigh0 reviewed (commented):
Can see the test runs fine in the ci job. One very minor readability thing

cailafinn committed:
Correct alg inputs test outputs

The algorithm is supposed to take two matrixworkspaces, rather than a
group. This empty cell workspace needs to be averaged, as it is a group.

RE #37120

cailafinn committed:
Refactor validation

RE #37120

cailafinn committed:
Clean up workspaces after test completes

RE #37120

jhaigh0 reviewed (commented):
Would it be easier for the algorithms in the test to use `StoreInADS=False` and then just assign the results to python variables?
If you want to use the ADS, is it easier to just clear the ADS in the `cleanup` method, rather than delete each test by name?

cailafinn commented:
> Would it be easier for the algorithms in the test to use `StoreInADS=False` and then just assign the results to python variables? If you want to use the ADS, is it easier to just clear the ADS in the `cleanup` method, rather than delete each test by name?

Ah, thanks for the heads up, that's a better solution. I was hesitant to just clear the ADS so it didn't clear other tests. But thinking about it, I don't think they run on the same instance of the framework.

cailafinn committed:
Refactor workspace preparation methods

RE #37120

jhaigh0 reviewed (dismissed):
Looks good to go, the test failure in the CI is unrelated
